pitop/robotics/navigation/core/measurement_scheduler.py

```diff
@@ -43,9 +49,10 @@ def loop(self, s, previous_time):
         self._new_measurement_event.set()
         self._new_measurement_event.clear()
 
-        s.enterabs(
-            current_time + self._measurement_dt,
-            1,
-            self.loop,
-            (s, current_time),
-        )
+        if self._continue_processing:
+            s.enterabs(
+                current_time + self._measurement_dt,
+                1,
+                self.loop,
```

Comment:
nice work getting all these threads to have proper exit conditions. unrelated but this use of recursion in this scheduler has caught my eye as something that could cause a stack overflow? maybe something to look into in future.